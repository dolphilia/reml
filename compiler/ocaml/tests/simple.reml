// Simple test file for Phase 1 Parser

module test.simple

use ::Core.Parse

let answer = 42

fn add(x, y) -> i64 = x + y

fn fact(n) -> i64 {
  if n <= 1 then 1 else n * fact(n - 1)
}

let result = add(10, 20)
