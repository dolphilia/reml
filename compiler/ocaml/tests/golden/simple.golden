module test.simple
use ::Core.Parse
let answer = int(42:base10)
fn add(x, y) -> i64 = binary(var(x) + var(y))
fn fact(n) -> i64 {block 1 stmts}
let result = call(var(add))[int(10:base10), int(20:base10)]
